src/main/java/com/example/cosmocats/exception/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,119 @@
+package com.example.cosmocats.exception;
+
+import jakarta.servlet.http.HttpServletRequest;
+
+import lombok.extern.slf4j.Slf4j;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.FieldError;
+
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+import org.springframework.web.context.request.ServletWebRequest;
+import org.springframework.web.context.request.WebRequest;
+
+
+@Slf4j
+@ControllerAdvice
+
+public class GlobalExceptionHandler {
+
+    @ExceptionHandler(MethodArgumentNotValidException.class)
+    @ResponseBody
+    public ResponseEntity<ErrorResponse> handleValidationExceptions(
+            MethodArgumentNotValidException ex, WebRequest request) {
+        
+        String path = ((ServletWebRequest) request).getRequest().getRequestURI();
+        String errorMessage = buildValidationErrorMessage(ex);
+        
+        ErrorResponse errorResponse = new ErrorResponse(
+                400,
+                "Bad Request",
+                errorMessage,
+                path
+        );
+        
+        log.warn("Validation error: {} - Path: {}", errorMessage, path);
+        return new ResponseEntity<>(errorResponse, HttpStatus.BAD_REQUEST);
```

Comment:
U don't need that - can just return the problem details, please checkout the code from lecture :) it's much more simpler